### PR TITLE
[INTERNAL] Allow the contrib package to be found by looking up an environment variable.

### PR DESCRIPTION
**Pull request by rrahn**
Needed to get a cleaner test environment for the CI builds via Jenkins

### CODE REVIEW

**Comment by rrahn** [comment]
rebuild

**Comment by rrahn** [comment]
Travis build error not related.

**Comment by rrahn** [comment]
rebuild

**Comment by h-2** [comment]
Anything I should review here or is this WIP?

**Comment by rrahn** [comment]
!rebuild

File: util/cmake/SeqAnContribs.cmake [hunk 1]
```diff
@@ -113,5 +119,27 @@ if (WIN32)
       endif (DEFINED SEQAN_CONTRIB_BASE)
 
     endforeach ()  # all contrib versions.
-endif (WIN32)
 
+    foreach (_BOOST_VERSION 1_65_1 1_65_0 1_64_0 1_63_0 1_62_0 1_61_0 1_60_0 1_59_0 1_58_0)
```
Comment (marehr) [inline util/cmake/SeqAnContribs.cmake]: I think we use `find(boost)` or so. There should be a variable to set the base path for finding boost.

This should make this code completely obsolete.

File: util/cmake/SeqAnContribs.cmake [hunk 2]
```diff
@@ -69,6 +69,12 @@ if (WIN32)
 		  set (SEQAN_CONTRIB_BASE "C:\\${_SEQAN_CONTRIB_DIR}-${CONTRIB_ARCH}")
 		endif ()
 	  endif (NOT DEFINED SEQAN_CONTRIB_BASE)
+    # Try to find in specific environment variable
```
Comment (marehr) [inline util/cmake/SeqAnContribs.cmake]: You could iterate over all possible directory paths.

In this code we always do the same:
* If SEQAN_CONTRIB_BASE is not known, yet
   * Look if the new path is existing
   * If not repeat

This is basically a for loop over all directories.

That can be  simplified:
```cmake
set(_paths "$ENV{ProgramW6432}" "$ENV{ProgramFiles}" "C:\\" "$ENV{SEQAN_WIN_CONTRIB_DIRECTORY}")

foreach(path $_paths)
        set(path "$path/${_SEQAN_CONTRIB_DIR}-${CONTRIB_ARCH}"
        if (IS_DIRECTORY "$path")
            set (SEQAN_CONTRIB_BASE "$path")
            break ()
        endif ()
endforeach()
```
(Might be no valid cmake syntax)

**Comment by rrahn** [comment]
!rebuild

**Pull request merged**